core/src/main/java/org/apache/carbondata/core/scan/complextypes/PrimitiveQueryType.java

```diff
@@ -53,6 +55,21 @@ public PrimitiveQueryType(String name, String parentname, int blockIndex,
     this.name = name;
     this.parentname = parentname;
     this.isDirectDictionary = isDirectDictionary;
+    this.isDictionary = true;
+  }
+
+
+  public PrimitiveQueryType(String name, String parentname, int blockIndex,
+      org.apache.carbondata.core.metadata.datatype.DataType dataType, int keySize,
+      Dictionary dictionary, boolean isDirectDictionary, boolean isDictionary) {
```

Comment:
I don't think it is required to have a separate constructor, if (dictionary == null && !isDirectDictionary) then it becomes nodictionary

Reply:
Done

Reply:
Done.